dotnet-engine/Yggdrasil.Engine/Flatbuffers.cs

```diff
@@ -0,0 +1,210 @@
+using System.Collections.ObjectModel;
+using System.Runtime.InteropServices;
+using Google.FlatBuffers;
+using yggdrasil.messaging;
+using Yggdrasil;
+using StrategyDefinition = Yggdrasil.StrategyDefinition;
+using Variant = Yggdrasil.Variant;
+using FlatVariant = yggdrasil.messaging.Variant;
+
+public static class Flatbuffers
+{
+    private static byte[] ReadBuffer(Buf buf)
+    {
+        ulong len = buf.len.ToUInt64();
```

Comment:
Think this is probably another thing that will work on 64 bit systems but not 32 bit ones